### PR TITLE
Issue #113 Logitech Media Server RockOn update for Rockstor

### PR DESCRIPTION
**Pull request by Hooverdan96**
Fixes #238 
Added new LogitechSqueezebox-7.9.4.json to RockOn repository
Using new docker image that also supports arm architecture
Update to root.json to reflect RockOn addition

### General information on project
This pull request proposes to add a new rock-on for the following project:
- name: Logitech Media Server (aka Squeezebox, slimserver) updated version
- website: https://mysqueezebox.com/index/Home
- description: this RockOn represents a new multi-architecture version for the Logitech Media Server, based on version 7.9.4

### Information on docker image
- docker image: https://hub.docker.com/r/doliana/logitech-media-server
- is an official docker image available for this project?: No. This one is periodically auto-build with new updates and supports ARM architecture

### Checklist
- [X] Passes [JSONlint](https://jsonlint.com) validation
- [X] Entry added to `root.json` in _alphabetical_ order (for new rock-on only)
- [X] `"description"` object lists and links to the docker image used
- [X] `"description"` object provides information on the image's particularities (advantage over another existing rock-on for the same project, for instance)
- [X] `"website"` object links to project's main website

### CODE REVIEW

**Comment by Hooverdan96** [comment]
Linked to Issue: #113

**Comment by phillxnet** [comment]
@Hooverdan96 This is great stuff, thanks. We should be able to begin working on the Rock-ons and Docs backlog shortly. Just getting our systems lined up ready to ease testing prs in both these repos, so not long now.

**Comment by Hooverdan96** [comment]
My apologies for pushing two more updates so quickly (I know you prefer a single push to review considering all the other stuff you're reviewing) ... after building another VM with the latest OpenSUSE release of Rockstor, I realized that my originally pushed version here contained an "older" version of the json file I had tested. I also recognized that this image does not support passing in a UID or PGID, so I made adjustments accordingly (highlighting in the tooltip that the config share needs r/w for UID=1000.
Finally, I restructured the description a bit for readability and added an icon link, too.

**Comment by FroggyFlox** [comment]
Hi @Hooverdan96 ,

I just had a look at your PR and it seems good! I only one problem with it, but that is completely independent from you: the use of host networking.
As evidenced in a recent PR (https://github.com/rockstor/rockon-registry/pull/227#issuecomment-680037260), this can create conflicts with the UI port when another rock-on is already set on using that same port (even without being installed). We've created an issue to keep track of it and solve the issue (https://github.com/rockstor/rockstor-core/issues/2209), but in the meantime, we may want to specify in the UI port description that one needs to use 9000 (and not something else) in order for the UI button to work. You can use the linked Jellyfin rock-on for an example if you'd like. Maybe we could also mention it in the "more_info" field?

**Comment by FroggyFlox** [comment]
@Hooverdan96 , I just realized I forgot to ask you the most important question:
Is host networking required in this rock-on? I suspect it is, but I just wanted to make sure.

**Comment by Hooverdan96** [comment]
@FroggyFlox, ah, yes, it unfortunately seems that way (similar to the Plex one). I tried to run it without but then I don't get the apple products to connect, while if I have net=host set, this seems to work.
I can add the reference to the more_info button, if that works for you.

Semi-unrelated .. and I can open a separate topic on this, the "more-info" is not really available until after the installation, can it be possible to have it visible before? Like in the jellyfin instance, if it's only available in the more info, it will be hard for a user to find out why a subsequent Docker install is not working. Or ... in the error message that you post after a failed install (go look in the logs), maybe one could add a reference to check out existing/installed rockons and possible port conflicts?
Sorry, just wanted to make a short remark on this, and now it's an entire paragraph.

**Comment by FroggyFlox** [comment]
> @FroggyFlox, ah, yes, it unfortunately seems that way (similar to the Plex one). I tried to run it without but then I don't get the apple products to connect, while if I have net=host set, this seems to work.

That makes sense, thanks for verifying that one out.



>  I can add the reference to the more_info button, if that works for you.

Sounds good! Could you also mention it in the UI port description, by any chance? The Jellyfin rock-on, for instance, will have it as:
```json
"description": "Jellyfin WebUI port. This MUST be 8096 due to current networking limitations",
```

> Semi-unrelated .. and I can open a separate topic on this, the "more-info" is not really available until after the installation, can it be possible to have it visible before? Like in the jellyfin instance, if it's only available in the more info, it will be hard for a user to find out why a subsequent Docker install is not working. Or ... in the error message that you post after a failed install (go look in the logs), maybe one could add a reference to check out existing/installed rockons and possible port conflicts?

That sounds like a good idea, indeed... Personally, I'm always concerned about cluttering, but that is far from it so it should be fine. It definitely is worth a try, at least, and see how it goes. I would thus say go for it and open a dedicated issue on the rockstor-core repo (as this is where changes would be required) linking to this issue and we can brainstorm on the best way to do it.

**Comment by Hooverdan96** [comment]
@FroggyFlox: I updated the WebUI portion with the corresponding comment. I decided then not to add the "More_Info" tag, since for this RockOn there wasn't one before. Hope, that's ok.
As I experiment with the LMS RockOn in the future, I will eventually propose the 8.x version as well and maybe I can also then figure out how to get around the net=host conundrum (maybe Linxuserver will eventually put one out, but who knows).

**Comment by FroggyFlox** [comment]
Sounds good, @Hooverdan96 ! If at any point you feel like the "More info" field could be beneficial for a rock-on (any rock-on) that doesn't have it, feel free to add it.

I'll try to spend some time today making some actual progress in our current rock-on backlog. Thanks a lot for all your contributions, that's incredibly helpful!

File: LogitechSqueezebox-7.9.4.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                  "label":"Config Storage"
+               },
+               "/srv/music":{
+                  "description":"Choose a Share for LMS content(your media). E.g.: create a Share called lms-music for this purpose alone.",
+                  "label":"Data Storage"
+               }
+            }
+         }
+      },
+      "description": "Logitech Media Server (aka Squeezebox, slimserver, etc.).<p>Stable version 7.9.4.</p><p>Based on custom docker image: <a href='https://hub.docker.com/r/doliana/logitech-media-server' target='_blank'>https://hub.docker.com/r/doliana/logitech-media-server</a>, available for amd64, arm32v7,arm64v8</p>",
```
Comment (FroggyFlox) [inline LogitechSqueezebox-7.9.4.json:46]: Minor typo:
```suggestion
      "description": "Logitech Media Server (aka Squeezebox, slimserver, etc.).<p>Stable version 7.9.4.</p><p>Based on custom docker image: <a href='https://hub.docker.com/r/doliana/logitech-media-server' target='_blank'>https://hub.docker.com/r/doliana/logitech-media-server</a>, available for amd64, arm32v7, and arm64v8</p>",
```

File: LogitechSqueezebox-7.9.4.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      "icon": "https://mysqueezebox.com/static/images/apple-touch-icon.png",
+      "more_info": "Logitech Media Server is a streaming audio server supported by Logitech (formerly Slim Devices), developed in particular to support their Squeezebox range of digital audio receivers. With the help of many plugins, Logitech Media Server can stream not only your local music collection, but content from many music services and internet radio stations to your players. See https://mysqueezebox.com/ and https://github.com/Logitech/slimserver for more information.",
```
Comment (FroggyFlox) [inline LogitechSqueezebox-7.9.4.json:48]: Add mention of the web UI port requirement:
```suggestion
      "more_info": "Logitech Media Server is a streaming audio server supported by Logitech (formerly Slim Devices), developed in particular to support their Squeezebox range of digital audio receivers. With the help of many plugins, Logitech Media Server can stream not only your local music collection, but content from many music services and internet radio stations to your players. See <a href='https://mysqueezebox.com/' target='_blank'>https://mysqueezebox.com/</a> and <a href='https://github.com/Logitech/slimserver' target='_blank'>https://github.com/Logitech/slimserver</a> for more information.<h4>Note about UI port</h4><p>Due to some current <a href='https://github.com/rockstor/rockstor-core/issues/2209' target='_blank'>limitations in rock-on networking</a>, the web UI port of this rock-on must be set as 9000 and will thus prevent the installation of other rock-ons needing this port. Alternatively, the web UI can be accessed manually at http://rockstor-ip:9000</p>",
```

**Review by FroggyFlox — CHANGES_REQUESTED** [review]

**Comment by FroggyFlox** [comment]
@Hooverdan96 ,

It all works well on my end! I just realized what you meant earlier with regards to the "More info" field; I misunderstood earlier, so my apologies for an erroneous answer at the time. I do believe we should mention the port requirement in the "More info" field in case the user would miss the info/tooltip during the install wizard. I thus added a suggestion accordingly. 

Let me know what you think of it.

**Comment by Hooverdan96** [comment]
@FroggyFlox: I accepted your suggestion. Makes sense to me, and since the More_info is "out of the way" is doesn't add to the clutter you were concerned about.

File: LogitechSqueezebox-7.9.4.json (hunk 1 above)
Comment (FroggyFlox) [reply LogitechSqueezebox-7.9.4.json:46]: @Hooverdan96 , any thought about this one?

**Comment by Hooverdan96** [comment]
yep, didn't see the typo piece, thanks for noticing.

**Review by FroggyFlox — APPROVED** [review]
@Hooverdan96 , nice submission! It all works as intended so I'm going to merge it now.

Thanks again for yet another nice rock-on!

**Pull request merged**